nuxeo-runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/avro/AvroSchemaStoreService.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     pierre
+ */
+package org.nuxeo.runtime.avro;
+
+import org.apache.avro.Schema;
+import org.apache.avro.message.SchemaStore;
+
+/**
+ * Holds avro schemas and allows to register them.
+ *
+ * @since 10.2
+ */
+public interface AvroSchemaStoreService extends SchemaStore {
```

Comment:
Isn't a bit risky to directly extends avro stuff ? shouldn't we prefer composition ?

Reply:
TBD

Reply:
can you give a short explanation why  ?

Reply:
In the use of Avro components, an instance of SchemaStore is needed, if we don't implement directly this interface there, we would have to return an instance of it, and then have another class which would implement it and the potential problem of the interface evolving would just be further.

Reply:
Since I don't know the real use cases to come I cannot answer this question.
The 3 topics tickets were separated into 3 tickets.
Plus, for now, the services and interfaces are located in nuxeo-runtime-stream, that does not depend on nuxeo-core, so implementations and tests are in nuxeo-core-io.
I believe for now it's better to have 3 services exposed by the same component, we'll see then what to do with them.